Makefile

```diff
@@ -1,28 +1,67 @@
-PROJECT_ROOT := src/
-VERSION = 3.2.0
+# Makefile used to create packages for mixer-tools. It doesn't assume
+# that the code is inside a GOPATH, and always copy the files into a
+# new workspace to get the work done. Go tools doesn't reliably work
+# with symbolic links.
+#
+# For historical purposes, it also works in a development environment
+# when the repository is already inside a GOPATH.
+
+GO_PACKAGE_PREFIX := github.com/clearlinux/mixer-tools
+
+.PHONY: gopath
+
+# Strictly speaking we should check if it the directory is inside an
+# actual GOPATH, but the directory structure matching is likely enough.
+ifeq (,$(findstring ${GO_PACKAGE_PREFIX},${PWD}))
+LOCAL_GOPATH := ${PWD}/.gopath
+export GOPATH := ${LOCAL_GOPATH}
+gopath:
+	@rm -rf ${LOCAL_GOPATH}/src
+	@mkdir -p ${LOCAL_GOPATH}/src/${GO_PACKAGE_PREFIX}
+	@cp -af * ${LOCAL_GOPATH}/src/${GO_PACKAGE_PREFIX}
+	@echo "Prepared a local GOPATH=${GOPATH}"
+else
+LOCAL_GOPATH :=
+GOPATH ?= ${HOME}/go
+gopath:
+	@echo "Code already in existing GOPATH=${GOPATH}"
+endif
+
+.PHONY: all compliant build install clean
 
 .DEFAULT_GOAL := all
+all: gopath compliant build
 
-CHANGES := $(shell go fmt ./... 2>&1)
+compliant: gopath
+	CHANGED_FILES=$$(go fmt ${GO_PACKAGE_PREFIX}/... 2>&1) ; \
+	if [ -n "$$CHANGED_FILES" ]; then \
+		echo -e "ERROR: go fmt had to fix the following files:\n$$CHANGED_FILES"; \
+		exit 1 ;\
+	fi
+	go vet ${GO_PACKAGE_PREFIX}/...
 
-# Ensure code is compliant
-compliant:
-	if [ "$(CHANGES)" ]; then echo -e "Error, go fmt ./... updated:\n$(CHANGES)"; exit 1 ; fi
-	go vet ./...
+build: gopath
+	go install ${GO_PACKAGE_PREFIX}/mixer
 
-install:
+install: gopath
 	test -d $(DESTDIR)/usr/bin || install -D -d -m 00755 $(DESTDIR)/usr/bin;
 	install -m 00755 $(GOPATH)/bin/mixer $(DESTDIR)/usr/bin/.
 	install -m 00755 pack-maker.sh $(DESTDIR)/usr/bin/mixer-pack-maker.sh
 	install -m 00755 superpack-maker.sh $(DESTDIR)/usr/bin/mixer-superpack-maker.sh
 	install -D -m 00644 yum.conf.in $(DESTDIR)/usr/share/defaults/mixer/yum.conf.in
 
+clean:
+	rm -f $(GOPATH)/bin/mixer
```

Comment:
GOPATH can be a colon separated list of directories.

Use ```go clean -i -x``` 

Should it clean up pack-maker.sh etc as well?

Reply:
Thanks for the go clean tip :-)

pack-maker.sh is part of the sources.

Reply:
I was rather thinking of removing the copy of` pack-maker.sh` from `/usr/bin` but perhaps the longer term fix is to add a `make uninstall` target?